ha/core/controllers/pcs/node_controller.py

```diff
@@ -71,7 +71,28 @@ def stop(self, node_id: str, timeout: int, **op_kwargs) -> dict:
             ([dict]): Return dictionary. {"status": "", "output": "", "error": ""}
                 status: Succeeded, Failed, InProgress
         """
-        raise HAUnimplemented("This operation is not implemented.")
+        check_cluster = op_kwargs.get("check_cluster") if op_kwargs.get("check_cluster") is not None else True
+        # Get the node_name (pvtfqdn) fron nodeid and raise exception if node_id is not valid
+        node_name = ConfigManager.get_node_name(node_id=node_id)
+        try:
+            timeout = const.NODE_STOP_TIMEOUT if timeout < 0 else timeout
+            node_status = self._system_health.get_node_status(node_id=node_id).get("status")
+            if node_status == HEALTH_STATUSES.OFFLINE.value:
+                Log.info(f"For stop node id {node_id}, Node already in offline state.")
+                status = f"Node with node id {node_id} is already in offline state."
+                return {"status": const.STATUSES.SUCCEEDED.value, "output": status, "error": ""}
+            elif node_status == NODE_STATUSES.POWEROFF.value or node_status == HEALTH_STATUSES.FAILED.value:
```

Comment:
can you verify this  ? in if condition you are checking against HEALTH_STATUSES and in this elif, check is against NODE_STATUSES. This is confusing.

Reply:
I think lets use HEALTH_STATUSES in all checks

Reply:
Used HEALTH_STATUSES. removed POWEROFF as we are not getting these status from system_health.